pkg/kube/inject/webhook_test.go

```diff
@@ -1148,6 +1148,21 @@ func TestParseInjectEnvs(t *testing.T) {
 			in:   "/inject/cluster/cluster1/network",
 			want: map[string]string{"ISTIO_META_CLUSTER_ID": "cluster1"},
 		},
+		{
+			name: "key-with-values-contain-slashes",
+			in:   "/inject/cluster/cluster2/rootpage/%2Ffoo%2Fbar",
+			want: map[string]string{"ISTIO_META_CLUSTER_ID": "cluster2", "ROOTPAGE": "/foo/bar"},
+		},
+		{
+			name: "no-predefined-kv-with-values-contain-slashes",
+			in:   "/inject/rootpage1/%2Ffoo%2Fbar/rootpage2/%2Ftic%2Ftac",
+			want: map[string]string{"ROOTPAGE1": "/foo/bar", "ROOTPAGE2": "/tic/tac"},
+		},
+		{
+			name: "no-predefined-kv-with-mixed-values",
+			in:   "/inject/rootpage1/%2Ffoo%2Fbar/rootpage2/%2Ftic%2Ftac%2F/rootpage3/something",
+			want: map[string]string{"ROOTPAGE1": "/foo/bar", "ROOTPAGE2": "/tic/tac/", "ROOTPAGE3": "something"},
+		},
```

Comment:
These tests are good, but I think we also need to also test this function in an http.Request context. See my other comment, above.